src/genetic_low_memory.c

```diff
@@ -36,47 +44,47 @@ static void tourney(uint16_t* population, unsigned int* firstParentIndex,
                     unsigned int* secondParentIndex,
                     float (*evaluationFunction)(float*)) {
 
-  float chosenIndexes[populationSize];
-  float bestFitness, secondbestFitness = 0;
+  float chosenIndexes[tournamentSelectionsSize];
+  float bestFitness = FLT_MAX;
+  float secondbestFitness = FLT_MAX;
 
   for (unsigned int i = 0; i < tournamentSelectionsSize; i++) {
 
     unsigned int index =
-        (unsigned int)(linear_congruential_random_generator() * populationSize);
-    int isAlreadyChosen = 1;
-
-    chosenIndexes[i] = index;
-    float parameters[dimensions];
-
-    if (i != 0) {
+        linear_congruential_random_generator() * populationSize;
```

Comment:
Don't we have a risk of selecting a parent outside of the population? If linear_congruential returns 1 you'll select the parent populationSize which is outside of the array? (Very unlikely I agree)